### DISCUSSION TITLE
How do i retrieve MessagesCollectionView object inside it's MessageColllectionViewCell

### DISCUSSION THREAD

**Discussion by Camelot93** [MessageKit/MessageKit#1680] [Q&A]
Hello I am trying to retrieve the parent `MessagesCollectionView` in my `MessageColllectionViewCell` i have created a custom `class` of type `MessageColllectionViewCell` with a `weak var parent`  of type `MessageColllectionViewCell?` and i have used `messagesCollectionView.register(customMessageCell.self)` to register it, the only problem is that i don't know where to give its attribute usually with `UICollectionViewCell` you can simply do `cell.parent = self` but I didn't find a protocol that allows access to attribute when a cell is created, how can I pass the value. Thank you

**Comment by martinpucik** [comment]
You can use `MessageContentCell` as a superclass for your custom Message cell. There is a method `func configure(with message: MessageType, at indexPath: IndexPath, and messagesCollectionView: MessagesCollectionView)` which will be called when populating the message cell content
